### PR TITLE
Hcl2 singular blocks

### LINKED ISSUE #8684: HCL2: Confusing or incomplete subnet_filter, ami_filter, security_group_filter, source_ami_filter, and vpc_filter
#### Overview of the Issue

When attempting to translate an existing JSON build configuration into the HCL2 syntax, I have a builders block in the old style like so:

```javascript
builders": [
    {
        "name": "base-ubuntu",
        "type": "amazon-ebs",
        "ami_name": "rhm/ubuntu/base-{{timestamp}}",

         // ...

        "subnet_filter": {
            "filters": {
                "tag:Name": "gateway-vpc-private*"
            },
            "random": true
        }

        // ...
]
```

But it is unclear how to translate the `subnet_filter` block, as `:` is not a valid HCL2 identifier character and identifiers cannot be quoted.

#### Reproduction Steps

1. Create an HCL2 configuration as in the example below
2. Attempt to run `packer build` on it

### Packer version

`Packer v1.5.1`

### Simplified Packer Buildfile

<details>
<summary>Example</summary>

```hcl
source "amazon-ebs" "base-ubuntu" {
  ami_name      = "rhm/ubuntu/base-${timestamp}"
  ami_users     = ["xxx"]
  region        = "us-east-1"
  instance_type = "m5.large"
  ebs_optimized = true
  ena_support   = true

  source_ami_filter {
    owners      = ["099720109477"]
    most_recent = true

    filters {
      virtualization-type = "hvm"
      name                = "ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-*"
      root-device-type    = "ebs"
    }
  }

  communicator     = "ssh"
  ssh_username     = "ubuntu"
  ssh_keypair_name = "gateway"
  ssh_agent_auth   = true
  ssh_interface    = "private_ip"

  subnet_filter {
    random = true
    filters {
      "tag:Name" = "gateway-vpc-private*"
    }
  }

  tags {
    Name     = "Ubuntu Base"
    Packer   = "true"
    BuiltAt  = "${isotime}"
    Platform = "Ubuntu"
    Release  = "18.04"
  }
}

build {
  sources = ["source.amazon-ebs.base-ubuntu"]

  provisioner "ansible" {
    playbook_file = "${pwd}/ansible/base.yml"
    galaxy_file   = "${pwd}/ansible/requirements.yml"
    groups        = ["base"]

    extra_arguments = [
      "--vault-password-file",
      ".vault-password",
    ]
  }
}
```
</details>

### Operating system and Environment details

<details>
<summary>screenfetch</summary>

```
 OS: 64bit Mac OS X 10.15.2 19C57
 Kernel: x86_64 Darwin 19.2.0
 Uptime: 1d 12m
 Packages: 130
 Shell: zsh 5.7.1
 Resolution: 1920x1080 ,1080x1920
 DE: Aqua
 WM: Quartz Compositor
 WM Theme: Blue
 Disk: 11G / 251G (19%)
 CPU: Intel Core i7-9750H @ 2.60GHz
 GPU: Intel UHD Graphics 630 / Radeon Pro 555X
 RAM: 8944MiB / 16384MiB
```
</details>

### Log Fragments and crash.log files

With example as above:

```
2020/02/03 09:26:42 [INFO] Packer version: 1.5.1 [go1.13.5 darwin amd64]
2020/02/03 09:26:42 Attempting to open config file: /Users/jlindsey/.packerconfig
2020/02/03 09:26:42 [WARN] Config file doesn't exist: /Users/jlindsey/.packerconfig
2020/02/03 09:26:42 Setting cache directory: /Users/jlindsey/projects/packer/packer_cache

Error: Invalid argument name
  on builds/base.pkr.hcl line 29, in source "amazon-ebs" "base-ubuntu":

  29:       "tag:Name" = "gateway-vpc-private*"

Argument names must not be quoted.

2020/02/03 09:26:42 Build debug mode: false
2020/02/03 09:26:42 Force build: false
2020/02/03 09:26:42 On error:
2020/02/03 09:26:42 Waiting on builds to complete...
==> Builds finished but no artifacts were created.
2020/02/03 09:26:42 [INFO] (telemetry) Finalizing.
  on builds/base.pkr.hcl line 29, in source "amazon-ebs" "base-ubuntu":
  29:       "tag:Name" = "gateway-vpc-private*"

Argument names must not be quoted.



==> Builds finished but no artifacts were created.
2020/02/03 09:26:43 waiting for all plugin processes to complete...
```

### PR DESCRIPTION
**Pull request by azr**
Allow to use singular blocks for map[string]string filters/setting/object.

I went for `name` `value` instead of `key` `value` because that's what [Terraform does](https://www.terraform.io/docs/providers/aws/d/subnet.html#name), also I think it makes more sense.

We keep both around because we want old systems to work the same.

Example:
```hcl
run_tags {
    run_tags_key = "value"
}

run_tag {
    name = "run_tag"
    value = "value"
}

spot_tags {
    spot_tags_key = "value"
}

spot_tag {
    name = "spot_tag"
    value = "value"
}

security_group_filter {
    filters {
        filters_key = "value"
    }
    filter {
        name = "filter_name"
        value = "value"
    }
}

source_ami_filter {
    filters {
        filters_key = "value"
    }
    filter {
        name = "filter_name"
        value = "value"
    }
}

subnet_filter {
    filters {
        filters_key = "value"
    }
    filter {
        name = "filter_name"
        value = "value"
    }
}

vpc_filter {
    filters {
        filters_key = "value"
    }
    filter {
        name = "filter_name"
        value = "value"
    }
}
``` 

# Why ?
this fixes #8684 and everything is laid out there

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -121,7 +121,6 @@ generate: install-gen-deps ## Generate dynamically generated code
 	@find post-processor common helper template builder provisioner -type f | xargs grep -l '^// Code generated' | xargs rm
 	go generate ./...
 	go fmt common/bootcommand/boot_command.go
-	go fmt command/plugin.go
```
Comment (azr) [inline Makefile]: I moved this to `scripts/generate-plugins.go`

File: cmd/struct-markdown/main.go [hunk 2]
```diff
@@ -117,6 +117,8 @@ func main() {
 				fieldType = `duration string | ex: "1h5m2s"`
 			case "config.Trilean":
 				fieldType = `boolean`
+			case "hcl2template.NameValues":
+				fieldType = `repeatable "name" & "value" object`
```
Comment (azr) [inline cmd/struct-markdown/main.go]: Here I'm trying to say in a super short sentence that this is a `[]{name string, value string }` but I'm not sure what's the best option for docs; do you have a suggestion ? May be I will try with `[]{name string, value string }`.

File: provisioner/shell/provisioner_test.go [hunk 3]
```diff
@@ -491,14 +491,14 @@ func TestProvisioner_RemoteFileDefaultsToScriptnnnn(t *testing.T) {
 		t.Fatalf("should not have error: %s", err)
 	}
 
-	remoteFileRegex := regexp.MustCompile("script_[0-9]{4}.sh")
+	remoteFileRegex := regexp.MustCompile("script_[0-9]{1,4}.sh")
```
Comment (azr) [inline provisioner/shell/provisioner_test.go]: This is to fix the following:
```
--- FAIL: TestProvisioner_RemoteFileDefaultsToScriptnnnn (0.00s)
    provisioner_test.go:497: remote_file did not default to script_nnnn.sh: "script_6.sh"
FAIL
```

Comment (SwampDragons) [reply provisioner/shell/provisioner_test.go]: I think I've seen this test failing elsewhere -- think you could make a separate PR for it so I can rebase the currently failing branches?

**Pull request merged**